devops/create_https_wss_endpoints_aws/main.py

```diff
@@ -0,0 +1,520 @@
+#!/usr/bin/env python3
+
+'''
+
+PURPOSE:
+
+create https/wss endpoints for a network
+
+USUAGE:
+
+$ python3 main.py <network_name>
+
+for example, the following cmd is used to create endpoints for testnet
+$ python3 main.py -n testnet
+
+
+'''
+
+import sys
+import json
+import os
+import argparse
+from collections import defaultdict
+
+import boto3
+import pprint
+from time import sleep
+import subprocess
+from dotenv import load_dotenv
+from timeit import default_timer as timer
+from datetime import timedelta
+
+
+
+
+pp = pprint.PrettyPrinter(indent=4)
+
+ap = argparse.ArgumentParser(description='parse the network type')
+ap.add_argument("-n", required=True, help="define network type")
+args = vars(ap.parse_args())
+
+current_work_path = os.path.dirname(os.path.realpath(__file__))
+network_config = current_work_path + '/' + args['n'] + '.json'
+
+
+
+# TO-DO: create the following dicts for each region
+# store target group arn, key: tg, value: arn of tg
+dict_tg_arn = dict()
+# store PREVIOUS instance id, key: tg, value: array of instance id
+dict_tg_instanceid = defaultdict(list)
+
+
+# store name of target group, key: tg_https, tg_wss, value: array of target group
+dict_tg_https_wss = defaultdict(list)
+
+
+# TO-DO: move to a helper class later
+def parse_network_config(param):
+    """ load the network configuration file, retrieve the value by its key """
+    with open(network_config, 'r') as f:
+        network_config_dict = json.load(f)
```

Comment:
it is easy to convert json to dict in python, so you'd consider a bit more structure data.